catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -0,0 +1,221 @@
+import ddf.security.Subject;
+import ddf.security.common.util.Security;
+
+/**
+ * Implementation of the {@link org.codice.ddf.migration.Migratable} interface used to migrate
+ * the current catalog framework's {@link Metacard}s.
+ */
+public class CatalogMigratableImpl extends AbstractMigratable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CatalogMigratableImpl.class);
+
+    private static final String DEFAULT_FAILURE_MESSAGE = "Catalog could not export metacards";
+
+    private static final String CATALOG_EXPORT_DIRECTORY = "org.codice.ddf.catalog";
+
+    private final CatalogFramework framework;
+
+    private final FilterBuilder filterBuilder;
+
+    private final MigrationFileWriter fileWriter;
+
+    private final CatalogMigratableConfig config;
+
+    private MigrationTaskManager taskManager;
+
+    /**
+     * Constructor.
+     *
+     * @param description   description of this migratable
+     * @param framework     framework used to retrieve the metacards to export
+     * @param filterBuilder builder used to create query filters
+     * @param fileWriter    object used to write metacards to disk
+     * @param config        export configuration information
+     */
+    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
+            @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
+            @NotNull CatalogMigratableConfig config) {
+        super(description, true);
+
+        notNull(framework, "CatalogFramework cannot be null");
+        notNull(filterBuilder, "FilterBuilder cannot be null");
+        notNull(fileWriter, "FileWriter cannot be null");
+        notNull(config, "Configuration object cannot be null");
+
+        this.framework = framework;
+        this.filterBuilder = filterBuilder;
+        this.fileWriter = fileWriter;
+        this.config = config;
+    }
+
+    /**
+     * Exports all the metacards currently stored in the catalog framework.
+     * <p>
+     * {@inheritDoc}
+     */
+    public MigrationMetadata export(Path exportPath) throws MigrationException {
+        taskManager = createTaskManager(config, createExecutorService(config));
+        config.setExportPath(exportPath.resolve(CATALOG_EXPORT_DIRECTORY));
+        fileWriter.createExportDirectory(config.getExportPath());
+
+        Collection<MigrationWarning> warnings = new ArrayList<>();
+        Map<String, Serializable> props = createMapWithNativeQueryMode();
+        Filter dumpFilter = filterBuilder.attribute(Metacard.ANY_TEXT)
+                .is()
+                .like()
+                .text("*");
+
+        QueryImpl exportQuery = createQuery(dumpFilter);
+        exportQuery.setPageSize(config.getExportQueryPageSize());
+        exportQuery.setRequestsTotalResultsCount(false);
+
+        QueryRequest exportQueryRequest = new QueryRequestImpl(exportQuery, props);
+
+        Subject subject = retrieveSystemSubject();
```

Comment:
This will allow any subject or even code with no subject to export all data.  It seems like we should require a subject that as an admin role before calling any export method.

https://github.com/codice/ddf/blob/ddf-2.8.0/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SubjectCommands.java#L63-L79